src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -647,40 +647,39 @@ public void SetMembrane(string membraneName)
         gui.UpdateMembraneButtons(Membrane.InternalName);
     }
 
-    public void SetRigidity(float rigidity)
+    public void SetRigidity(int rigidity)
     {
-        if (Math.Abs(Rigidity - rigidity) < MathUtils.EPSILON)
+        int intRigidity = (int)Math.Round(Rigidity * 10);
+
+        if (intRigidity == rigidity)
             return;
 
-        var cost = (int)(Math.Abs(rigidity - Rigidity) / 2 * 100);
+        int cost = Math.Abs(rigidity - intRigidity) * Constants.MEMBRANE_RIGIDITY_COST_PER_STEP;
 
-        if (cost > 0)
+        if (cost > MutationPoints)
         {
-            if (cost > MutationPoints)
-            {
-                rigidity = Rigidity + (rigidity < Rigidity ? -MutationPoints : MutationPoints) * 2 / 100.0f;
-                cost = MutationPoints;
-            }
+            gui.UpdateRigiditySlider(intRigidity, MutationPoints);
```

Comment:
I think it would be nice if this went as far as possible, because I can think of a situation where the user presses a specific point on the bar, and that should move it as close as possible to the clicked point.